plugin/src/main/java/org/opensearch/ml/rest/RestMLRegisterModelAction.java

```diff
@@ -107,8 +107,10 @@ MLRegisterModelRequest getRequest(RestRequest request) throws IOException {
         MLRegisterModelInput mlInput = modelName == null
             ? MLRegisterModelInput.parse(parser, loadModel)
             : MLRegisterModelInput.parse(parser, modelName, version, loadModel);
-        if (mlInput.getUrl() != null && !allowModelUrl) {
-            throw new IllegalArgumentException("Don't allow uploading model via url.");
+        if (mlInput.getUrl() != null && !isModelUrlAllowed) {
+            throw new IllegalArgumentException(
+                "To upload custom model user needs to enable allow_model_url settings. Otherwise please use opensearch pre-trained models."
```

Comment:
I think according to nate's suggestion we need to change the settings name to: `allow_registering_model_via_url` ?